syntax = "proto3";

package email;

service EmailService {
  rpc SendEmail (EmailRequest) returns (EmailResponse) {}
  rpc QueryEmails (EmailQueryRequest) returns (stream EmailResponse) {}
}

message EmailRequest {
  string subject = 1;
  string message = 2;
  string emailTo = 3;
  string emailFrom = 4;
}

message EmailResponse {
  bool success = 1;
  string message = 2;
}

message EmailQueryRequest {
  int64 start_time = 1;
  int64 end_time = 2;
  string email_address = 3;
}